pages/extensions/amp_dependencies/__init__.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+import uuid
+from sets import Set
+
+from grow import extensions
+from grow.documents import document, static_document
+from grow.extensions import hooks
+
+class AmpDependencies(object):
+    def __init__(self, doc):
+        self._doc = doc
+        self._pod = doc.pod
+        # Used to determine where to print all dependencies
+        self._placeholder = '{}'.format(uuid.uuid4())
+        # Stores registered paths
+        self._dependencies = []
```

Comment:
Should this not be a set, add may be called multiple times for the same dependency
Still the question would remain what to do if different versions are requested of the same dependency

Reply:
It has been converted to a set later but is now one initially 🙂